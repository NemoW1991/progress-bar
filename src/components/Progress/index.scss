.progress {
  position: relative;
  span {
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    color: black;
  }

  &--color-purple[value] {
    &::-webkit-progress-value {
      background-image: -webkit-linear-gradient(left,#623aa2, #f97794 100% );
    }
  }
  &--color-pink[value]{
    &::-webkit-progress-value {
      background-color: #801143 ;
    }
  }
}  

progress[value] {
  width: 35em;
  height: 2em;
  border-radius: 10px;
  margin: 10px;
  box-shadow: 1px 1px 4px rgba( 0, 0, 0, 0.2 );

  &::-webkit-progress-bar {
    background-color: white;
    border-radius: 10px;
  }

  &::-webkit-progress-value {
    border-radius: 10px;
    background-size: 200% 100%;
  }

  @media only screen and(max-width: 1024px) {
    width: 30em;
  }
  @media only screen and(max-width: 768px) {
    width: 25em;
  }
  @media only screen and(max-width: 480px) {
    width: 20em;
  }
  @media only screen and(max-width: 320px) {
    width: 18em;
  }
}