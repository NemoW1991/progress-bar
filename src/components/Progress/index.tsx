import React from 'react';
import './index.scss';
import {getPercentage} from '../../lib/util';

const Progress: React.FC<{
  readonly bar?: number;
  readonly limit?: number;
}> = ({
  bar = 0,
  limit = 100
}) => {
  const progressPercentage = getPercentage(bar, limit);

  return (
    <div className='progress'>
      <progress id="index" value={bar} max={limit} className={progressPercentage>=100 ? 'progress--color-pink': 'progress--color-purple'}>{bar}</progress>
      <span>{progressPercentage}%</span>
    </div>
  )
}

export default Progress;