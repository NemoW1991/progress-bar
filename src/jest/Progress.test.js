import { shallow } from 'enzyme';
import Progress from '../components/Progress';

const mockData = {bar: 30, limit: 200};
describe('<Progress>', () => {
  
  let wrapper;
  beforeEach(() => wrapper = shallow(
    <Progress 
      bar = {mockData.bar}
      limit = {mockData.limit}
    />
  ));

  it('renders without crashing', () => {
    expect(wrapper.length).toBe(1);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should has correct value', () => {
    expect(wrapper.find('progress').prop('value')).toBe(30)
  });

  it('should has correct max value', () => {
    expect(wrapper.find('progress').prop('max')).toBe(200)
  })

})