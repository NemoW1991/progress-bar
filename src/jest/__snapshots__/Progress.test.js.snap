// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Progress> should render correctly 1`] = `ShallowWrapper {}`;
